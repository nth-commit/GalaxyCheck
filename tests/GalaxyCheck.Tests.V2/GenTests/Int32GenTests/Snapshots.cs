using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int32GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int32().Between(0, 10000) },
                { "Between_Default", Gen.Int32() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
