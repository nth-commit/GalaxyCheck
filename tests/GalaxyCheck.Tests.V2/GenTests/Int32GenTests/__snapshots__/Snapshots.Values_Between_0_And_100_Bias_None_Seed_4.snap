[
  31,
  64,
  2,
  73,
  13,
  98,
  75,
  94,
  17,
  83,
  38,
  47,
  9,
  23,
  88,
  53,
  49,
  3,
  14,
  53,
  69,
  4,
  63,
  7,
  52,
  38,
  49,
  58,
  12,
  87,
  32,
  64,
  66,
  91,
  60,
  69,
  71,
  94,
  26,
  39,
  28,
  13,
  79,
  57,
  76,
  57,
  97,
  77,
  82,
  92,
  90,
  14,
  85,
  86,
  66,
  13,
  48,
  73,
  3,
  40,
  43,
  86,
  73,
  83,
  68,
  47,
  72,
  68,
  72,
  89,
  11,
  7,
  65,
  28,
  47,
  76,
  73,
  33,
  56,
  90,
  77,
  72,
  73,
  68,
  69,
  83,
  66,
  26,
  67,
  32,
  2,
  44,
  74,
  34,
  98,
  99,
  27,
  32,
  29,
  60
]
